typescript/sdk/src/ism/metadata/aggregation.ts

```diff
@@ -9,7 +24,54 @@ export interface AggregationIsmMetadata {
 const RANGE_SIZE = 4;
 
 // adapted from rust/agents/relayer/src/msg/metadata/aggregation.rs
-export class AggregationIsmMetadataBuilder {
+export class AggregationIsmMetadataBuilder
+  implements
+    MetadataBuilder<
+      WithAddress<AggregationIsmConfig>,
+      DerivedHookConfigWithAddress
+    >
+{
+  protected logger = rootLogger.child({
+    module: 'AggregationIsmMetadataBuilder',
+  });
+
+  constructor(protected readonly base: BaseMetadataBuilder) {}
+
+  async build(
+    message: DispatchedMessage,
+    context: {
+      ism: WithAddress<AggregationIsmConfig>;
+      hook: DerivedHookConfigWithAddress;
+      dispatchTx: TransactionReceipt;
+    },
+    maxDepth = 10,
+  ): Promise<string> {
+    assert(maxDepth > 0, 'Max depth reached');
+    const promises = await Promise.allSettled(
```

Comment:
Just confirming you intended to use `allSettled` here vs `all`?

Reply:
yep, we can accommodate some failures

Reply:
Optional: Might need to add a timeout here?